docs/index.adoc

```diff
@@ -218,139 +218,180 @@ libStatic.buildGetter('my/folder', {
-        getCleanPath: request => request.pathParams.resourcePath
+// Set up lib-static:
+const getStatic = libStatic.buildGetter(
+    {
+        root: 'static',
+        getCleanPath: request => request.pathParams.libRouterPath
     }
-));
+);
+libRouter.get(
+    `/static/{libRouterPath:.+}`,
+    request => getStatic(request)
+);
+
 
-// <webappURL> and <webappURL>/
-libRouter.get( `/`, function (request) {
+// Instead of exports.get, libRouter takes over:
+libRouter.get(`/`, request => {
+    const assetsRootUrl = `${request.contextPath}/static`;
     return {
         body: `
-            <html>
-              <head>
-                <title>It still works</title>
-                <link   rel="stylesheet"
-                        type="text/css"
-                        href="${request.contextPath}/getResource/styles.css"
-                />
-              </head>
-
-              <body>
-                  <h1>It still works!</h1>
-                  <img src="${request.contextPath}/getResource/logo.jpg" />
-                  <script src="${request.contextPath}/getResource/js/myscript.js"></script>
-              </body>
-            </html>
-            `
+<html>
+    <head>
+        <title>Webapp</title>
+        <link rel="stylesheet" type="text/css" href="${assetsRootUrl}/styles.css" />
+    </head>
+
+    <body>
+        <h1>My webapp</h1>
+        <img src="${assetsRootUrl}/images/my-logo.jpg" />
+    </body>
+</html>
+        `
     };
 });
 ----
 
-{zwsp} +
-
 [NOTE]
 ====
-It might seem tempting to just let the links in the HTML (`${request.contextPath}/getResource/...`) start with `getResource/`. That looks neater and simpler and could just let the browser append them as relative links, and resolve its requests to `<webappURL>/getResource/...` etc.
+Another gotcha: in this example, `request.contextPath` is used as a simple but stable way to get the webapp's own root URL, aka `<webappRoot>`. Other approaches may be needed for other contexts (e.g. vhosting, etc), but either way:
 
-However, in XP, the webapp will respond to both `<webappURL>` and `<webappURL>/` - note the trailing slash, which makes the relative link behave in two different ways, only one of which is right. And adding a `/` at the beginning, `/getResource/...`, is of course no solution either, just an absolute path from the domain root.
+Since `<webappRoot>/static` is directly below the webapp's root, *why resolve the webapp's root URL, instead of just using a relative path*? Why not just use `static/styles.css` and `static/images/my-logo.jpg`?
 
-Prefixing with `request.contextPath` solves it in this case. Your mileage may vary.
+The XP webapp engine makes the controller respond identically to both `<webappRoot>` and `<webappRoot>/`. That trailing slash makes for two different contexts and resolves to two different relative addresses, so one stable, explicit value is needed.
```

Comment:
IMO, it is an unfortunate bug in XP webapp (not yet reported?)

Reply:
There's still no issue?

Also, what might be the time scale for that to be fixed? It seems like something that might be considered a breaking change, so, XP8? If so, better keep this in the docs for now.

Reply:
https://github.com/enonic/xp/issues/7087